// @qwik.dev/router@2.0.0-alpha.1/lib/middleware/cloudflare-pages/index.mjs
// packages/qwik-router/src/middleware/cloudflare-pages/index.ts
import { getNotFound } from "@qwik-router-not-found-paths";
import { isStaticPath } from "@qwik-router-static-paths";
import { _deserialize, _serialize, _verifySerializable } from "@qwik.dev/core/internal";
import { setServerPlatform } from "@qwik.dev/core/server";
import {
  _TextEncoderStream_polyfill,
  mergeHeadersCookies,
  requestHandler
} from "../request-handler/index.mjs";
function createQwikRouter(opts) {
  if (opts.qwikCityPlan && !opts.qwikRouterConfig) {
    console.warn("qwikCityPlan is deprecated. Use qwikRouterConfig instead.");
    opts.qwikRouterConfig = opts.qwikCityPlan;
  } else if (!opts.qwikRouterConfig) {
    throw new Error("qwikRouterConfig is required.");
  }
  try {
    new globalThis.TextEncoderStream();
  } catch (e) {
    globalThis.TextEncoderStream = _TextEncoderStream_polyfill;
  }
  const qwikSerializer = { _deserialize, _serialize, _verifySerializable };
  if (opts.manifest) {
    setServerPlatform(opts.manifest);
  }
  async function onCloudflarePagesFetch(request, env, ctx) {
    try {
      const url = new URL(request.url);
      if (isStaticPath(request.method, url)) {
        return env.ASSETS.fetch(request);
      }
      const useCache = url.hostname !== "127.0.0.1" && url.hostname !== "localhost" && url.port === "" && request.method === "GET";
      const cacheKey = new Request(url.href, request);
      const cache = useCache ? await caches.open("custom:qwikrouter") : null;
      if (cache) {
        const cachedResponse = await cache.match(cacheKey);
        if (cachedResponse) {
          return cachedResponse;
        }
      }
      const serverRequestEv = {
        mode: "server",
        locale: void 0,
        url,
        request,
        env: {
          get(key) {
            return env[key];
          }
        },
        getWritableStream: (status, headers, cookies, resolve) => {
          const { readable, writable } = new TransformStream();
          const response = new Response(readable, {
            status,
            headers: mergeHeadersCookies(headers, cookies)
          });
          resolve(response);
          return writable;
        },
        getClientConn: () => {
          return {
            ip: request.headers.get("CF-connecting-ip") || "",
            country: request.headers.get("CF-IPCountry") || ""
          };
        },
        platform: {
          request,
          env,
          ctx
        }
      };
      const handledResponse = await requestHandler(serverRequestEv, opts, qwikSerializer);
      if (handledResponse) {
        handledResponse.completion.then((v) => {
          if (v) {
            console.error(v);
          }
        });
        const response = await handledResponse.response;
        if (response) {
          if (response.ok && cache && response.headers.has("Cache-Control")) {
            ctx.waitUntil(cache.put(cacheKey, response.clone()));
          }
          return response;
        }
      }
      const notFoundHtml = isStaticPath(request.method || "GET", url) ? "Not Found" : getNotFound(url.pathname);
      return new Response(notFoundHtml, {
        status: 404,
        headers: { "Content-Type": "text/html; charset=utf-8", "X-Not-Found": url.pathname }
      });
    } catch (e) {
      console.error(e);
      return new Response(String(e || "Error"), {
        status: 500,
        headers: { "Content-Type": "text/plain; charset=utf-8", "X-Error": "cloudflare-pages" }
      });
    }
  }
  return onCloudflarePagesFetch;
}
var createQwikCity = createQwikRouter;
export {
  createQwikCity,
  createQwikRouter
};
