// @qwik.dev/router@2.0.0-alpha.1/lib/middleware/aws-lambda/index.mjs
// packages/qwik-router/src/middleware/aws-lambda/index.ts
import { createQwikRouter as createQwikRouterNode } from "@qwik.dev/router/middleware/node";
function createQwikRouter(opts) {
  if (opts.qwikCityPlan && !opts.qwikRouterConfig) {
    console.warn("qwikCityPlan is deprecated. Use qwikRouterConfig instead.");
    opts.qwikRouterConfig = opts.qwikCityPlan;
  } else if (!opts.qwikRouterConfig) {
    throw new Error("qwikRouterConfig is required.");
  }
  try {
    const { router, staticFile, notFound } = createQwikRouterNode({
      render: opts.render,
      qwikRouterConfig: opts.qwikRouterConfig,
      manifest: opts.manifest,
      static: {
        cacheControl: "public, max-age=31557600"
      },
      getOrigin(req) {
        if (process.env.IS_OFFLINE) {
          return `http://${req.headers.host}`;
        }
        return null;
      }
    });
    const fixPath = (pathT) => {
      if (opts.qwikRouterConfig.trailingSlash) {
        const url = new URL(pathT, "http://aws-qwik.local");
        if (url.pathname.includes(".", url.pathname.lastIndexOf("/"))) {
          return pathT;
        }
        if (!url.pathname.endsWith("/")) {
          return url.pathname + "/" + url.search;
        }
      }
      return pathT;
    };
    const handle = (req, res) => {
      req.url = fixPath(req.url);
      staticFile(req, res, () => {
        router(req, res, () => {
          notFound(req, res, () => {
          });
        });
      });
    };
    return { fixPath, router, staticFile, notFound, handle };
  } catch (err) {
    throw new Error(err.message);
  }
}
var createQwikCity = createQwikRouter;
export {
  createQwikCity,
  createQwikRouter
};
